lib/prima_auth0_ex/middleware/require_permission.ex

```diff
@@ -0,0 +1,42 @@
+defmodule PrimaAuth0Ex.Middleware.RequirePermission do
+  @moduledoc """
+  Absinthe middleware that ensure the permission is included in the current security context.
+  """
+
+  require Logger
+
+  alias Absinthe.Resolution
+  alias PrimaAuth0Ex.Plug.CreateSecurityContext.Context
+
+  @behaviour Absinthe.Middleware
+
+  @impl true
+  def call(
+        %{context: %Context{dry_run: dry_run, permissions: permissions}} = resolution,
+        required_permission
+      ) do
+    permissions
+    |> has_required_permission?(required_permission)
+    |> then(fn has_required_permissions? ->
+      # On dry-run, log a warning if token doesn't have required permission
+      # but permissions was not nil (i.e. it was supposed to work)
+      if not has_required_permissions? and permissions != nil and dry_run do
+        Logger.warn("Received invalid token", required_permission: required_permission)
```

Comment:
shouldn't this simply go into the resolve function (in the corresponding match?)